genie-web/src/main/java/com/netflix/genie/web/apis/rest/v3/controllers/JobRestController.java

```diff
@@ -263,15 +263,15 @@ public JobRestController(
     @ResponseStatus(HttpStatus.ACCEPTED)
     public ResponseEntity<Void> submitJob(
         @Valid @RequestPart("request") final JobRequest jobRequest,
-        @RequestPart("attachment") final MultipartFile[] attachments,
+        @RequestPart(value = "attachment", required = false) @Nullable final MultipartFile[] attachments,
         @RequestHeader(value = FORWARDED_FOR_HEADER, required = false) @Nullable final String clientHost,
         @RequestHeader(value = HttpHeaders.USER_AGENT, required = false) @Nullable final String userAgent,
         final HttpServletRequest httpServletRequest
     ) throws GenieException, GenieCheckedException {
         log.info(
             "[submitJob] Called multipart method to submit job: {}, with {} attachments",
             jobRequest,
-            attachments.length
+            attachments == null ? 0 : attachments.length
```

Comment:
Would this be an issue if the attachment is actually null? I assume the `handleSubmitJob` at line `277` can take null just fine.

Reply:
This specific line is for the array so it shouldn't matter if the internal attachments are null. The `handle` method was already re-used between both so should be ok but we'll test all this with our repro's before we actually release.